connect/api/src/main/java/org/apache/kafka/connect/sink/SinkRecord.java

```diff
@@ -57,6 +73,105 @@ public TimestampType timestampType() {
         return timestampType;
     }
 
+    /**
+     * Get the original topic for this sink record, corresponding to the topic of the Kafka record before any
+     * {@link org.apache.kafka.connect.transforms.Transformation Transformation}s were applied. This should be used by
+     * sink tasks for any internal offset tracking purposes (that are reported to the framework via
+     * {@link SinkTask#preCommit(Map)} for instance) rather than {@link #topic()}, in order to be compatible with
+     * transformations that mutate the topic name.
+     * <p>
+     * This method was added in Apache Kafka 3.6. Sink connectors that use this method but want to maintain backward
+     * compatibility in order to be able to be deployed on older Connect runtimes should guard the call to this method
+     * with a try-catch block, since calling this method will result in a {@link NoSuchMethodException} or
+     * {@link NoClassDefFoundError} when the sink connector is deployed to Connect runtimes older than Kafka 3.6.
+     * For example:
+     * <pre>{@code
+     * String originalTopic;
+     * try {
+     *     originalTopic = record.originalTopic();
+     * } catch (NoSuchMethodError | NoClassDefFoundError e) {
+     *     originalTopic = record.topic();
+     * }
+     * }
+     * </pre>
+     * <p>
+     * Note that sink connectors that do their own offset tracking will be incompatible with SMTs that mutate the topic
+     * name when deployed to older Connect runtimes.
```

Comment:
Nit: language tweak, clarification on unsupported Connect runtimes
```suggestion
     * Note that sink connectors that do their own offset tracking will be incompatible with SMTs that mutate topic
     * names when deployed to older Connect runtimes that do not support this method.
```